src/less-watch-compiler.js

```diff
@@ -114,9 +114,15 @@ function init(){
       } else {
         // f is a new file or changed
         var importedFile = false;
+        var filename = f.substring(lessWatchCompilerUtils.config.watchFolder.length+1)
         for (var i in fileimportlist){
           for (var k in fileimportlist[i]){
-            if (f.substring(lessWatchCompilerUtils.config.watchFolder.length+1) == fileimportlist[i][k]){
+            var hasExtension = fileimportlist[i][k].substring(fileimportlist[i][k].lastIndexOf('/') + 1).indexOf('.') != -1;
```

Comment:
What is this doing here?

Reply:
A checks to see if the filename has an extension, if not it assumes that it is a .less file.
See point 3 in #55, it has a link to the less documentation.